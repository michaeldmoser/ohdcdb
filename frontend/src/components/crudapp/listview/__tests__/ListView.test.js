import { render } from 'testing/library';
import { ListView, ListItem } from '../index';

describe('Render the list view', () => {
    it('should render with defaults', () => {
        const query = {
            data: [
                {
                    id: 1,
                    title: 'Title 1',
                    additionalInfo: 'Additional info 1',
                },
                {
                    id: 2,
                    title: 'Title 2',
                    additionalInfo: 'Additional info 2',
                },
                {
                    id: 3,
                    title: 'Title 3',
                    additionalInfo: 'Additional info 3',
                },
            ],
            isLoading: false,
        };

        const listview = render(
            <ListView listName='List of properties' query={query}>
                {({ title, additionalInfo }) => (
                    <>
                        {title} {additionalInfo}
                    </>
                )}
            </ListView>
        );

        expect(listview).toMatchSnapshot();
    });

    it('should no matching results when there are no records', () => {
        const query = {
            data: null,
            isLoading: false,
        };

        const listview = render(
            <ListView listName='List of properties' query={query}>
                {() => {}}
            </ListView>
        );

        expect(listview).toMatchSnapshot();
    });

    it('should show a loading message', () => {
        const query = {
            data: null,
            isLoading: true,
        };

        const listview = render(
            <ListView listName='List of properties' query={query}>
                {() => {}}
            </ListView>
        );

        expect(listview).toMatchSnapshot();
    });
});
