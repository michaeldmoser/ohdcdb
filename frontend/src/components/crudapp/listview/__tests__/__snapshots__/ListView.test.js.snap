// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the list view should no matching results when there are no records 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="list-view"
      >
        <header
          class="visually-hidden"
        >
          <h4>
            List of properties
          </h4>
        </header>
        <p>
          No matching records could be found.
        </p>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="list-view"
    >
      <header
        class="visually-hidden"
      >
        <h4>
          List of properties
        </h4>
      </header>
      <p>
        No matching records could be found.
      </p>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render the list view should render with defaults 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="list-view"
      >
        <header
          class="visually-hidden"
        >
          <h4>
            List of properties
          </h4>
        </header>
        <ul
          class="list-group list-group-flush"
        >
          <li
            class="nav-item list-group-item"
          >
            <a
              class="lh-sm"
              href="/1"
            >
              Title 1
               
              Additional info 1
            </a>
          </li>
          <li
            class="nav-item list-group-item"
          >
            <a
              class="lh-sm"
              href="/2"
            >
              Title 2
               
              Additional info 2
            </a>
          </li>
          <li
            class="nav-item list-group-item"
          >
            <a
              class="lh-sm"
              href="/3"
            >
              Title 3
               
              Additional info 3
            </a>
          </li>
        </ul>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="list-view"
    >
      <header
        class="visually-hidden"
      >
        <h4>
          List of properties
        </h4>
      </header>
      <ul
        class="list-group list-group-flush"
      >
        <li
          class="nav-item list-group-item"
        >
          <a
            class="lh-sm"
            href="/1"
          >
            Title 1
             
            Additional info 1
          </a>
        </li>
        <li
          class="nav-item list-group-item"
        >
          <a
            class="lh-sm"
            href="/2"
          >
            Title 2
             
            Additional info 2
          </a>
        </li>
        <li
          class="nav-item list-group-item"
        >
          <a
            class="lh-sm"
            href="/3"
          >
            Title 3
             
            Additional info 3
          </a>
        </li>
      </ul>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render the list view should show a loading message 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="list-view"
      >
        <header
          class="visually-hidden"
        >
          <h4>
            List of properties
          </h4>
        </header>
        <p>
          Loading results...
        </p>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="list-view"
    >
      <header
        class="visually-hidden"
      >
        <h4>
          List of properties
        </h4>
      </header>
      <p>
        Loading results...
      </p>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
